File: src/com/designpatterns/creational/builder/Customer.java
package com.designpatterns.creational.builder;

public class Customer {

    private final Chef chef;

    public Customer(Chef chef) {
        this.chef = chef;
    }

    public void orderAndEatPizza() {
        // Get pepperoni pizza and eat.
        Pizza pizza = chef.getPepperoniPizza();
        System.out.println("PepperoniPizza:" + pizza.eatPizza());
        // Get pepperoni pizza and eat.
        Pizza pizza1 = chef.getSupremeVegPizza();
        pizza1.eatPizza();
        System.out.println("SupremeVegPizza:" + pizza1.eatPizza());
    }
}

File: src/com/designpatterns/creational/builder/kotlin/Customer.kt
package com.designpatterns.creational.builder.kotlin

class Customer(private val chef: Chef) {

    fun orderAndEatPizza() {
        // Get pepperoni pizza and eat.
        val pizza = chef.pepperoniPizza
        println("PepperoniPizza:" + pizza.eatPizza())

        // Get pepperoni pizza and eat.
        val pizza1 = chef.supremeVegPizza
        pizza1.eatPizza()
        println("SupremeVegPizza:" + pizza1.eatPizza())
    }
}